ext/cl_khr_command_buffer.asciidoc

```diff
@@ -0,0 +1,1875 @@
+  _command_buffer_, and _kernel_ are not the same.
+
+`CL_INVALID_EVENT_WAIT_LIST` is replaced with:
+
+* `CL_INVALID_SYNC_POINT_WAIT_LIST_KHR` if _sync_point_wait_list_ is `NULL` and
+  _num_sync_points_in_wait_list_ is > 0, or _sync_point_wait_list_ is not
+  `NULL` and _num_sync_points_in_wait_list_ is 0, or if
+  synchronization-point objects in _sync_point_wait_list_ are not valid
+  synchronization-points.
+
+New errors:
+
+* `CL_INVALID_COMMAND_BUFFER_KHR` if _command_buffer_ is not a valid
+  command-buffer.
+
+* `CL_INVALID_VALUE` if values specified in _properties_ are not valid
+
+* `CL_INVALID_OPERATION` if _command_buffer_ has been finalized.
+
+* `CL_INVALID_VALUE` if _mutable_handle_ is not `NULL`.
+
+* `CL_INVALID_OPERATION` if the device associated with _command_queue_ does not
+  support `CL_COMMAND_BUFFER_CAPABILITY_KERNEL_PRINTF_KHR` and _kernel_ contains
+  a printf call.
+
+* `CL_INVALID_OPERATION` if the device associated with _command_queue_ does not
+  support `CL_COMMAND_BUFFER_CAPABILITY_DEVICE_SIDE_ENQUEUE_KHR` and _kernel_
+  contains a kernel-enqueue call.
+
+==== Add new section "Section 5.X.4 - Command Buffer Queries"
+
+The function
+indexterm:[clGetCommandBufferInfoKHR]
+[source]
+----
+cl_int clGetCommandBufferInfoKHR(
+    cl_command_buffer_khr command_buffer,
+    cl_command_buffer_info_khr param_name,
+    size_t param_value_size,
+    void* param_value,
+    size_t* param_value_size_ret);
+----
+Queries information about a command-buffer.
+
+_command_buffer_ Specifies the command-buffer being queried.
+
+_param_name_ Specifies the information to query.
+
+_param_value_size_ Specifies the size in bytes of memory pointed to by
+_param_value_. This size must be ≥ size of return type as described in the table
+below. If _param_value_ is `NULL`, it is ignored.
+
+_param_value_ A pointer to memory where the appropriate result being queried is
+returned. If _param_value_ is `NULL`, it is ignored.
+
+_param_value_size_ret_ Returns the actual size in bytes of data being queried by
+_param_value_. If _param_value_size_ret_ is `NULL`, it is ignored.
+
+The list of supported _param_name_ values and the information returned in
+_param_value_ by *clGetCommandBufferInfoKHR* is described in the table below.
+
+.*clGetCommandBufferInfoKHR* values
+[cols=",,",options="header",]
+|====
+| *cl_command_buffer_info_khr*
+| *Return Type*
+| *Description*
+
+| *CL_COMMAND_BUFFER_NUM_QUEUES_KHR*
+| cl_uint
+| The number of command-queues specified when _command_buffer_ was created.
+
+| *CL_COMMAND_BUFFER_QUEUES_KHR*
+| cl_command_queue[]
+| Return the list of command-queues specified when the _command_buffer_ was
+  created.
+
+| *CL_COMMAND_BUFFER_REFERENCE_COUNT_KHR*
+| cl_uint
+| Return the _command_buffer_ reference count.
+
+  The reference count returned with `CL_COMMAND_BUFFER_REFERENCE_COUNT_KHR`
+  should be considered immediately stale. It is provided for identifying
+  memory leaks.
```

Comment:
```suggestion
| *CL_COMMAND_BUFFER_REFERENCE_COUNT_KHR* footnote:[{fn-reference-count-usage}]
| cl_uint
| Return the _command_buffer_ reference count.
```

Will only work if https://github.com/KhronosGroup/OpenCL-Docs/pull/712 is merged first.

Reply:
I think this makes sense, but I'll hold off committing this suggestion until #712 is merged.